config/s6-overlay/s6-rc.d/haproxy/run

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1091
+
+# HAProxy service for s6-overlay
```

Comment:
This run file is emulating the upstream [haproxy service file](https://sources.debian.org/src/haproxy/3.0.11-1/admin/systemd/haproxy.service.in) + our overlay service file `config/confd/templates/haproxy.conf.tmpl` (deleted in this PR).

The important parts:
- `KillMode=mixed` handled by the signal handler function below
- Extra `EnvironmentFile`s handled inline below
- `LimitNOFILE=1048576` handled inline below
- `TimeoutStopSec` moved to `config/s6-overlay/scripts/entry.sh`
- `SuccessExitStatus=143` handled inline below